jira/utils/__init__.py

```diff
@@ -36,13 +36,18 @@ class CaseInsensitiveDict(dict):
     def __init__(self, *args, **kw):
         super(CaseInsensitiveDict, self).__init__(*args, **kw)
 
-        self.itemlist = {}
-        for key, value in super(CaseInsensitiveDict, self).items():
-            if key != key.lower():
-                self[key.lower()] = value
-                self.pop(key, None)
+#        self.itemlist = {}
+#        for key, value in super(CaseInsensitiveDict, self).items():
+#            if key != key.lower():
+#                self[key.lower()] = value
+#                self.pop(key, None)
 
         # self.itemlist[key.lower()] = value
+        itemlist = {}
+        for key, value in super(CaseInsensitiveDict, self).items():
+            itemlist[key.lower()] = value
```

Comment:
Why not use a list of keys to remove and remove them in subsequent for loop?

Reply:
thank you. Yes, that would be clearer actually and appears to work fine too:
```
    def __init__(self, *args, **kw):
        super(CaseInsensitiveDict, self).__init__(*args, **kw)

        upper_keys_list = []
        for key in super(CaseInsensitiveDict, self).keys():
            if key != key.lower():
                upper_keys_list.append(key)

        for upper_key in upper_keys_list:
            self[upper_key.lower()] = self[upper_key]
            self.pop(upper_key, None)
```